src/Controller/ProductFinderController.php

```
@@ -149,5 +158,153 @@ public function chatSearch(#[MapRequestPayload] ChatRequestDto $chatRequest): Js
+        // In a production app, use a proper file storage service (e.g., S3, or Symfony's public dir with unique names).
+        $tempDir = $this->projectDir . '/public/uploads/temp_images/';
+        if (!is_dir($tempDir) && !mkdir($tempDir, 0777, true) && !is_dir($tempDir)) {
+            // @codeCoverageIgnoreStart
+            // This is hard to test reliably without more complex setup
+            return $this->json(new ChatResponseDto(false, null, 'Could not create temporary directory for image upload.'), 500);
+            // @codeCoverageIgnoreEnd
+        }
+        $imageFileName = uniqid('img_', true) . '.' . $imageFile->guessExtension();
+        try {
+            //$imageFile->move($tempDir, $imageFileName);
+        } catch (\Exception $e) {
```

PHPStan (2.1.15)

Dead catch - Exception is never thrown in the try block.


(catch.neverThrown)